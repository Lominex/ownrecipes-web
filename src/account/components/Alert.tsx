import { defineMessages, useIntl } from 'react-intl';

import Alert from '../../common/components/Alert';
import { ReducerMeta } from '../../common/store/GenericReducerType';

export interface ILoginAlert {
  reducerMeta: ReducerMeta;
}

const LoginAlert: React.FC<ILoginAlert> = ({ reducerMeta }: ILoginAlert) => {
  const { formatMessage } = useIntl();

  const messages = defineMessages({
    errorTitle: {
      id: 'login.alert.error_title',
      description: 'Fail to login header',
      defaultMessage: 'Login failed',
    },
    errorMessage: {
      id: 'login.alert.incorrect_credentials',
      description: 'Fail to login message',
      defaultMessage: 'Incorrect username or password.',
    },
  });

  const { error } = reducerMeta;

  return (
    <>
      {error && (
        <Alert severity='danger' title={formatMessage(messages.errorTitle)}>
          {formatMessage(messages.errorMessage)}
        </Alert>
      )}
    </>
  );
};

export default LoginAlert;
