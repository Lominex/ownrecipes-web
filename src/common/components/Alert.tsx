import { Alert as BootstrapAlert } from 'react-bootstrap';

import '../css/alert.css';

import Icon from './Icon';
import P from './P';

export interface IAlertProps {
  severity: 'danger' | 'info';
  title: string;
  className?: string;

  children: React.ReactNode;
}

const Alert: React.FC<IAlertProps> = ({ severity, title, className, children }: IAlertProps) => (
  <BootstrapAlert variant={severity} className={className}>
    <BootstrapAlert.Heading>
      {severity === 'info'   && <Icon icon='info-circle' size='2x' className='alert-icon' />}
      {severity === 'danger' && <Icon icon='exclamation-diamond' size='2x' className='alert-icon' />}
      {title}
    </BootstrapAlert.Heading>
    {typeof children === 'string' && (
      <P className='alert-message mb-0'>{children}</P>
    )}
    {typeof children !== 'string' && (
      <div className='alert-message'>{children}</div>
    )}
  </BootstrapAlert>
);

export default Alert;
