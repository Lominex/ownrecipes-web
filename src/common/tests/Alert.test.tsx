import Alert from '../components/Alert';
import renderer from 'react-test-renderer';

const TITLE = 'This is some alert';
const MSG = 'This is some message';
const MSG_JSX = (
  <span className='some-span-class'>
    <p className='some-p-class'>{MSG}</p>
  </span>
);

test('Alert component test', () => {
  const component = renderer.create(
    <Alert severity='danger' title={TITLE}>
      {MSG}
    </Alert>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Alert component jsx test', () => {
  const component = renderer.create(
    <Alert severity='danger' title={TITLE}>
      {MSG_JSX}
    </Alert>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Alert component info test', () => {
  const component = renderer.create(
    <Alert severity='info' title={TITLE}>
      {MSG}
    </Alert>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
