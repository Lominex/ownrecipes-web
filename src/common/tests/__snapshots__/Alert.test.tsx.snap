// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component info test 1`] = `
<div
  className="fade alert alert-info show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    <i
      className="bi bi-info-circle-fill size-2x alert-icon"
    />
    This is some alert
  </div>
  <p
    className="body1 alert-message mb-0"
  >
    This is some message
  </p>
</div>
`;

exports[`Alert component jsx test 1`] = `
<div
  className="fade alert alert-danger show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    <i
      className="bi bi-exclamation-diamond-fill size-2x alert-icon"
    />
    This is some alert
  </div>
  <div
    className="alert-message"
  >
    <span
      className="some-span-class"
    >
      <p
        className="some-p-class"
      >
        This is some message
      </p>
    </span>
  </div>
</div>
`;

exports[`Alert component test 1`] = `
<div
  className="fade alert alert-danger show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    <i
      className="bi bi-exclamation-diamond-fill size-2x alert-icon"
    />
    This is some alert
  </div>
  <p
    className="body1 alert-message mb-0"
  >
    This is some message
  </p>
</div>
`;
